sixtools/AD_Handler.py

```diff
@@ -0,0 +1,68 @@
+from databroker.assets.handlers_base import HandlerBase
+import os.path
+
+
+class HDF5SingleHandler(HandlerBase):
+    '''Handler for hdf5 data stored 1 image per file.
+    Parameters
+    ----------
+    fpath : string
+        filepath
+    template : string
+        filename template string.
+    filename : string
+        filename
+    key : string
+        the 'path' inside the file to the data set.
+    frame_per_point : float
+        the number of frames per point.
+    '''
+    specs = {'AD_HDF5_SINGLE'} | HandlerBase.specs
+
+    def __init__(self, fpath, template, filename, key, frame_per_point=1):
+        self._path = os.path.join(fpath, '')
+        self._fpp = frame_per_point
+        self._template = template
+        self._filename = filename
+        self._key = key
+
+    def _fnames_for_point(self, point_number):
+        start = int(point_number * self._fpp)
+        stop = int((point_number + 1) * self._fpp)
+        for j in range(start, stop):
+            yield self._template % (self._path, self._filename, j)
+
+    def __call__(self, point_number):
+        ret = []
+        import h5py
+        for fn in self._fnames_for_point(point_number):
+            f = h5py.File(fn, 'r')
```

Comment:
Can you do this with a context manager to ensure that the file gets closed?

Reply:
Is this really necessary to implement in this temporary solution? perhaps this comment is better in the data broker PR (here)[https://github.com/NSLS-II/databroker/pull/375]. They are looking to have this implemented as soon as possible and this does not seem like something that means this won't work.

Reply:
not being careful about managing when and how we open hdf5 files has bitten us else where, we should try to get things right even in the "temporary" solution.

Reply:
@tacaswell can you please confirm that you are in agreement to punt the context manager to the databroker PR? we can update this one once we have a solution

Reply:
We should use the context manager here to make sure we are tidy about file I/O.

```python
with h5py.File(fn, 'r') as f:
    # rest of the code is the same
```

Reply:
added the context manager and tested, so this is done.